internal/checker/terraform/rules/gcp/google_compute_instance_disk_encryption.rego

```diff
@@ -0,0 +1,86 @@
+# METADATA
+# title: "Verify Google Compute Instance Encryption with Customer Supplied Encryption Keys (CSEK) for Critical VM Disks"
+# description: "This policy ensures that disks for critical Virtual Machines (VMs) are encrypted using Customer Supplied Encryption Keys (CSEK). CSEK provides an added layer of security for sensitive data."
+# related_resources:
+# - https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_instance
+# custom:
+#   id: CB_TFGCP_080
+#   severity: HIGH
+package lib.terraform.CB_TFGCP_080
+
+import future.keywords.in
+
+isvalid(block) {
+	block.Type == "resource"
+	"google_compute_instance" in block.Labels
+}
+
+resource[resource] {
+	some block in pass
+	resource := concat(".", block.Labels)
+}
+
+resource[resource] {
+	some block in fail
+	resource := concat(".", block.Labels)
+}
+
+pass[resource] {
+	some resource in input
+	isvalid(resource)
+	some block in resource.Blocks
+	block.Type == "disk_encryption_key"
+}
+
+pass[resource] {
+	some resource in input
+	isvalid(resource)
+	some block in resource.Blocks
+	block.Type == "boot_disk"
+	some innerblock in block.Blocks
+	innerblock.Type == "disk_encryption_key_raw"
+}
+
+pass[resource] {
+	some resource in input
+	isvalid(resource)
+	some block in resource.Blocks
+	block.Type == "boot_disk"
+	some innerblock in block.Blocks
+	innerblock.Type == "disk_encryption_key"
+	innerblock.Attributes.disk_encryption_key_raw != null
+	innerblock.Attributes.kms_key_self_link != null
+}
+
```

Comment:
remove line 55 - 63
it is identical to lin 44 - 53

Reply:
Noted on this one. Thank you for checking.